kernel/include/ksched.h

```diff
@@ -43,11 +43,13 @@ void z_unpend_thread_no_timeout(struct k_thread *thread);
 struct k_thread *z_unpend1_no_timeout(_wait_q_t *wait_q);
 int z_pend_curr(struct k_spinlock *lock, k_spinlock_key_t key,
 	       _wait_q_t *wait_q, k_timeout_t timeout);
+int z_pend_curr_with_arch_lock(_wait_q_t *wait_q, k_timeout_t timeout);
```

Comment:
@andyross I know that you're trying to _reduce_ the number of APIs here, and here I am adding another one...